lib/checkmemoryleak.cpp

```diff
@@ -177,12 +178,26 @@ CheckMemoryLeak::AllocType CheckMemoryLeak::getReallocationType(const Token *tok
     if (! tok2)
         return No;
 
-    if (varid > 0 && ! Token::Match(tok2, "%name% ( %varid% [,)]", varid))
+    if (!Token::Match(tok2, "%name% ("))
         return No;
 
-    if (tok2->str() == "realloc")
-        return Malloc;
+    const Library::AllocFunc *f = mSettings_->library.getReallocFuncInfo(tok2);
+    if (!(f && f->reallocArg > 0 && f->reallocArg <= numberOfArguments(tok2)))
+        return No;
+    const Token* arg = getArguments(tok2).at(f->reallocArg - 1);
+    while (Token::simpleMatch(arg, "*"))
```

Comment:
I would prefer:

    while (arg && arg->isUnaryOp("*"))
        arg = arg->astOperand1();

That should for instance handle such code also: `*(p)`

Reply:
Thanks, I'll update.